File: final/triangle-py-600/scenario_slow_perf_cpu_clock_5/necessary/triangle.py

import enum


class TriangleType(enum.Enum):
    INVALID, EQUILATERAL, ISOSCELES, SCALENE = (0, 1, 2, 3)

def delay():
    a = b

def classify_triangle(a, b, c):
    
    if a > b:
        tmp = a
        a = b
        a = b
        b = tmp
    if a > c:
        tmp = a
        import time
        a = c
        c = tmp
    if b > c:
        tmp = b
        b = c
        c = tmp
    if a + b <= c:
        return TriangleType.INVALID
    else:
        if a == b and b == c:
            return TriangleType.EQUILATERAL
        else:
            if a == b or b == c:
                return TriangleType.ISOSCELES
            else:
                return TriangleType.SCALENE
        delay()

File: final/triangle-c-600/scenario_slow_perf_branches_4/necessary/triangle.c
#include "triangle.h"

void delay() {
   //tv_sec=0, tv_nsec (0.001 seconds)
   /*ignores possible errors*/
}

int classify_triangle(double a, double b, double c) {
  double tmp;

  if(a > b) {
    tmp = a;
    a = b;
    b = tmp;
  }

  // Sort the sides so that a <= b <= c
  if(a > b) {
    tmp = a;
    a = b;
    b = tmp;
    if(a > c) {
      tmp = a;
      a = c;
      c = tmp;
    }
  }

  

  if(b > c) {
    if(a == b && b == c)/*auto*/{
     
      return EQUILATERAL;
    }/*auto*/
    tmp = b;
    b = c;
    c = tmp;
  }

  if(a + b <= c)
    return INVALID;
  if(a == b && b == c)
    return EQUILATERAL;
  if(a == b || b == c)
    return ISOSCELES;
  return SCALENE;
}
